bravado/requests_client.py

```diff
@@ -185,18 +218,55 @@ def check_for_exceptions(self, response):
         except Exception as e:
             add_response_detail_to_errors(e)
 
+    def build_timeout(self, result_timeout):
+        """
+        Build the appropriate timeout object to pass to `session.send(...)`.
+
+        :param result_timeout: timeout that was passed into `result(..)`.
+        :return: timeout
+        :rtype: float or tuple(connect_timeout, timeout)
+        """
+        timeout = None
+        has_service_timeout = 'timeout' in self.misc_options
+        has_result_timeout = result_timeout is not None
```

Comment:
I actually think Fido shouldn't have a default here since it just turns around and calls [concurrent.futures.Future::result(timeout=None)](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Future.result) which does a good job of defining what passing `None` means (wait indefinitely).

Changing the default in fido is a breaking change which I'm not ready to take on atm.